### PR TITLE
Standardize output method to printf in polar_decoder_common.cc

### PR DESCRIPTION
**Pull request by SWITCHin2**
## Description

Refactor output method in polar_decoder_common.cc to utilize printf instead of cout for consistency.

## Related Issue

3rd Sub-task of #4765 

## Testing Done

Yes

## Checklist

- [ x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [ ] I have squashed my commits to have one significant change per commit. 
- [ ] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x ] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Review by marcusmueller — CHANGES_REQUESTED** [review]
Hi! The problem is to change C++ std::cout to the logging facilies, not to convert it to C89 `printf`.

**Comment by marcusmueller** [comment]
Also, C-style casts are a bit frowned upon.

You will want to look at existing logging code in GNU Radio, for example https://github.com/gnuradio/gnuradio/blob/6466dad0e79fa0bb4d7276117127b893ed43e2b3/gr-fec/lib/dummy_encoder_impl.cc#L58 .

Please don't forget to sign your commit! We won't merge it otherwise.

When you're done, please also remove the `#includes` of `cstdio` and `iostream`.

**Comment by marcusmueller** [comment]
Hm, thinking of it, maybe this really isn't a logging problem, but can be solved as one:

1. instead of `<cstdio>` and `<iostream>`, include `<spdlog/fmt/fmt.h>`
2. Convert the `std::cout <<` and `printf` stuff to a consistent use of `fmt::print()`. 

You can find the syntax for how to print stuff with `fmt` on https://fmt.dev/latest/syntax.html ; it's a lot like Python's format strings!

File: gr-fec/lib/polar_decoder_common.cc [hunk 1]
```diff
@@ -168,11 +168,11 @@ void polar_decoder_common::print_pretty_llr_vector(const float* llr_vec) const
 {
     for (int row = 0; row < block_size(); row++) {
         // FIXME this is an interesting mixture of iostream and stdio
```
Comment (marcusmueller) [inline gr-fec/lib/polar_decoder_common.cc:170]: You'd want to remove this comment once it's consistently doing `fmt::print`, as well!

**Review by marcusmueller — CHANGES_REQUESTED** [review]

File: gr-fec/lib/polar_decoder_common.cc (hunk 1 above)
Comment (SWITCHin2) [reply gr-fec/lib/polar_decoder_common.cc:170]: Ok, then refactoring to `fmt::print`

Comment (marcusmueller) [reply gr-fec/lib/polar_decoder_common.cc:170]: thank you so much :)

**Pull request closed without merging**